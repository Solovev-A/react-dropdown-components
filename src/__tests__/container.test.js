import React from 'react';
import { render } from "@testing-library/react";
import 'jest-styled-components';

import Container from './../lib/components/container';

describe('Container component', () => {
    describe('renders correctly', () => {
        test('without props passed', () => {
            const { container } = render(<Container />);
            expect(container.firstChild).toMatchSnapshot();
        });

        test('when disabled', () => {
            const { container } = render(<Container disabled={true} />);
            expect(container.firstChild).toMatchSnapshot();
        });

        test('when dropdown is open at top', () => {
            const { container } = render(<Container dropdownPosition="top" isDropdownOpen={true} />);
            expect(container.firstChild).toMatchSnapshot();
        });

        test('when dropdown is open at bottom', () => {
            const { container } = render(<Container dropdownPosition="bottom" isDropdownOpen={true} />);
            expect(container.firstChild).toMatchSnapshot();
        });
    });
});