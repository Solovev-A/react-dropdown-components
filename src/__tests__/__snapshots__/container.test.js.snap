// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component renders correctly when disabled 1`] = `
.c0 {
  position: relative;
  min-height: calc(1.5em + .75rem + 2px) !important;
  width: 100%;
  margin: 0;
  padding-left: .375rem;
  padding-right: .375rem;
  padding-bottom: .375rem;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #fff;
  border: 1px solid #ced4da;
  border-radius: .25rem;
  cursor: pointer;
  pointer-events: none;
  background-color: #e9ecef;
}

.c0:focus-within {
  box-shadow: 0 0 0 0.2rem rgb(38 143 255 / 50%);
}

<div
  class="c0"
  disabled=""
/>
`;

exports[`Container component renders correctly when dropdown is open at bottom 1`] = `
.c0 {
  position: relative;
  min-height: calc(1.5em + .75rem + 2px) !important;
  width: 100%;
  margin: 0;
  padding-left: .375rem;
  padding-right: .375rem;
  padding-bottom: .375rem;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #fff;
  border: 1px solid #ced4da;
  border-radius: .25rem;
  cursor: pointer;
  border-bottom-right-radius: 0;
  border-bottom-left-radius: 0;
  border-bottom-color: #fff;
}

.c0:focus-within {
  box-shadow: 0 0 0 0.2rem rgb(38 143 255 / 50%);
}

<div
  class="c0"
/>
`;

exports[`Container component renders correctly when dropdown is open at top 1`] = `
.c0 {
  position: relative;
  min-height: calc(1.5em + .75rem + 2px) !important;
  width: 100%;
  margin: 0;
  padding-left: .375rem;
  padding-right: .375rem;
  padding-bottom: .375rem;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #fff;
  border: 1px solid #ced4da;
  border-radius: .25rem;
  cursor: pointer;
  border-top-right-radius: 0;
  border-top-left-radius: 0;
  border-top-color: #fff;
}

.c0:focus-within {
  box-shadow: 0 0 0 0.2rem rgb(38 143 255 / 50%);
}

<div
  class="c0"
/>
`;

exports[`Container component renders correctly without props passed 1`] = `
.c0 {
  position: relative;
  min-height: calc(1.5em + .75rem + 2px) !important;
  width: 100%;
  margin: 0;
  padding-left: .375rem;
  padding-right: .375rem;
  padding-bottom: .375rem;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #fff;
  border: 1px solid #ced4da;
  border-radius: .25rem;
  cursor: pointer;
}

.c0:focus-within {
  box-shadow: 0 0 0 0.2rem rgb(38 143 255 / 50%);
}

<div
  class="c0"
/>
`;
